javascript/implementation/LogBackedStore.ts

```diff
@@ -202,11 +214,6 @@ export class LogBackedStore implements Store {
 
     async claimChain(medallion: Medallion, chainStart: ChainStart, actorId?: ActorId): Promise<ClaimedChain> {
         await this.ready;
-        const unlockingFunction = await this.memoryLock.acquireLock();
```

Comment:
This function is only called from addBundle where we have already acquired the lock. Before removing this, claimChain and addBundle would just fight for the file. Let me know if I am overlooking something though.

Reply:
Yeah that makes sense you only need to get the memory lock in one place, and if claimChain is becoming a private function then it can't be called directly.